src/gwt/src/org/rstudio/studio/client/common/vcs/AskPassManager.java

```diff
@@ -78,10 +79,21 @@ public void onAskPass(final AskPassEvent e)
             
             askpassPending_ = true;
             
-            globalDisplay.promptForPassword(
-                  "Password",
-                  e.getPrompt(),
+            String prompt = e.getPrompt();
```

Comment:
Keying off of a visible UI string to change fundamental behavior makes me... uneasy. That sort of thing will bite us if we every try to localize the UI, and makes the code less self-documenting.

Plus the fact this class is called `AskPassManager` and can now prompt for something that isn't a password seems unexpected (though, to be fair, it was already kinda doing that).

I think this means that rstudioapi::askForPassword will now show a regular text field instead of a password field if the prompt contains "username"?

If separating this behavior entirely (e.g. separate messages and handler class) is overkill, might want to document this quirk in `rstudioapi` and also in SessionAskPass.cpp / askPassword().

Reply:
> Plus the fact this class is called AskPassManager and can now prompt for something that isn't a password seems unexpected (though, to be fair, it was already kinda doing that).

I definitely agree it's weird, but ultimately that's what a non-interactive `GIT_ASKPASS` program has to do. That is, they are programs that read a prompt from `stdin`, and then return an appropriate value (username or password) based on that prompt. It definitely feels unnatural to me, but ... it is what it is. There may be cleaner ways to handle this, but this is at least a straightforward solution.

Perhaps it would be cleaner if we made that decision on the session side; e.g. here:

https://github.com/rstudio/rstudio/blob/0691592b8743cbe291317af331a0adc5526f1ca0/src/cpp/session/modules/SessionAskPass.cpp#L100-L104

and then just had the UI respond based appropriately based on whether it "looks like" we're prompting for a username or not.

Reply:
Got it, makes sense.